src/Core/Binding/ServerConnection.cs

```diff
@@ -25,7 +25,7 @@ public abstract class ServerConnection
     internal static readonly ServerConnectionSettings DefaultSettings = new(true);
     
     public string Id { get; }
-    public ServerConnectionSettings Settings { get; }
+    public ServerConnectionSettings Settings { get; set; }
```

Comment:
Why was the setter added?

Reply:
Because in the repository we have TryUpdateSettingsById, which requires to add the settings to an existing ServerConnection model

Reply:
I don't know if we should do that. Feels like a thread safety issue. The model can be modified from another thread. Could we make it immutable? Would that have big implications on other parts of the code?

Reply:
In the repository, I am reading the file and then retrieving the ServerConnection model for the provided Id. I then have to update the settings: 
![image](https://github.com/user-attachments/assets/dac23354-e702-4c5e-b11e-d74ac971f5c2)


I either allow modification of the settings or I need to create a new instance of the ServerConnection (which is immutable), then remove it from the list, then re-add the new instance. This feels a lot more error-prone and unexpected.

I could make instead the IsSmartNotificationsEnabled editable (add a setter for it). Also I don't think we should be too paranoid about the multi-threading. Because all the write operations to the connections should only happen via the UI. And the UI will be blocked while anything in the connection is modified